var/spack/repos/builtin/packages/nvhpc/package.py

```diff
@@ -89,24 +90,38 @@ class Nvhpc(Package):
     provides('blas',        when='+blas')
     provides('lapack',      when='+lapack')
     provides('mpi',         when='+mpi')
+    patch('makelocalrc_path.patch')
 
     def install(self, spec, prefix):
         # Enable the silent installation feature
         os.environ['NVHPC_SILENT'] = "true"
         os.environ['NVHPC_ACCEPT_EULA'] = "accept"
         os.environ['NVHPC_INSTALL_DIR'] = prefix
 
+        version_prefix = join_path(self.prefix, 'Linux_%s' %
+                                   self.spec.target.family,
+                                   self.version)
+        flags = ' -q'
+
         if spec.variants['install_type'].value == 'network':
+            local_dir = version_prefix + '/share_objects'
             os.environ['NVHPC_INSTALL_TYPE'] = "network"
-            os.environ['NVHPC_INSTALL_LOCAL_DIR'] = \
-                "%s/%s/%s/share_objects" % \
-                (prefix, 'Linux_%s' % spec.target.family, self.version)
+            os.environ['NVHPC_INSTALL_LOCAL_DIR'] = local_dir
+            flags = flags + ' -net ' + local_dir
         else:
             os.environ['NVHPC_INSTALL_TYPE'] = "single"
-
         # Run install script
         os.system("./install")
 
+        # Update localrc to use Spack gcc
+        if spec.satisfies('%gcc'):
+            prefix = Prefix(join_path(version_prefix, 'compilers'))
+            script = join_path(prefix.bin, 'makelocalrc')
+            flags = flags + ' -gcc %s -gpp %s -g77 %s -x %s' % (
+                self.compiler.cc, self.compiler.cxx,
+                self.compiler.f77, prefix.bin)
```

Comment:
I gave this PR a try and it turns out that the patch (https://github.com/spack/spack/pull/24654/files#diff-cf482bcc3e73486f18429a2cf2092deee184fe53c6d55cc464e325ebb044e838) is definitely required at least on systems that have compilers in modules (I tried this on a cray system). Since `self.compiler.cc` is `cc` (another wrapper) at least in my case (likewise `cxx` and `f77`) prepending `/bin:/usr/bin` ends up having `cc` pick up the default system `g++` even if spack has otherwise ensured that the correct `g++` would be used. This makes `makelocalrc` misdetect the compilers.

The patch should also be updated to apply for different platforms.

@pjh40 do you have plans to pick this up again?